### PR TITLE
S5 Refactor Backend

### LINKED ISSUE #156: DB and api small refactor
Database schema
- plurals (questions, reviews etc..)
- just id instead of other jank
- course table uses course code only

API
- /courses/:code/questions or reviews/:id
 
Routes
- /courses/:code/questions or reviews/:id

### PR DESCRIPTION
**Pull request by DarkPurple141**
Lodging now to keep things granular.

See if you can find any bugerinos.

Resolves #156.

### CODE REVIEW

**Comment by NunoDasNeves** [comment]
You need to branch off of dev! look at this commit change hnng. oh welll

**Comment by DarkPurple141** [comment]
Yeah my bad.

File: frontend/tests/unit/Feed.spec.js [hunk 1]
```diff
@@ -29,10 +29,10 @@ describe('Feed.vue Reviews', () => {
 describe('Feed.vue Questions', () => {
   before(function() {
     this.items = [
-      { id: 1, title: 'hey', body: 'wow', likes: 2, published: new Date().toDateString() },
-      { id: 2, title: 'blurggh', body: 'wowzers', likes: 3, published: new Date().toDateString() },
-      { id: 3, title: 'blurggh', body: 'woawders', likes: 3, published: new Date().toDateString() },
-      { id: 4, title: 'blurggh', body: 'wdiajdw', likes: 3, published: new Date().toDateString() }
+      { id: '1', title: 'hey', body: 'wow', likes: 2, published: new Date().toDateString() },
```
Comment (DarkPurple141) [inline frontend/tests/unit/Feed.spec.js]: Maybe id should be a number I dunno. Where is it getting affected most? When it comes out of the URL?

**Comment by DarkPurple141** [comment]
I can't approve this cos I'm the owner.

**Pull request merged**